Updates to email/slack job notifications (#4347)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/job-notifications.md

---
title: "Job notifications"
id: "job-notifications"
description: "Set up notifications in dbt Cloud to receive email or Slack alerts about job run status."
---

Set up notifications in dbt Cloud to receive email or Slack alerts when a job run succeeds, fails, or is cancelled.

## Email notifications

You can receive email alerts by configuring the dbt Cloud notification settings. As a **user**, you can set up email notifications for yourself. As an **admin**, you can set up notifications for yourself and other team members. Refer to [Users and licenses](/docs/cloud/manage-access/seats-and-users) for details on license types that are eligible for email notifications.

### Configure email notifications

1. From the gear menu in the upper right corner, choose **Notification settings**.
1. By default, the **Notification email** option is set to the email address that's in [your dbt Cloud personal profile](https://cloud.getdbt.com/settings/profile). 

>     If you're an **admin**, you can choose a different email address. Select the **Notification email** dropdown and choose another address from the list. The list includes internal users with access to the account and also external emails that have been added. 

Reviewer: Can we specify *account* admin?

Author: for sure!

>     - To add an external email address, select the **Notification email** dropdown and choose **Add external email**. After you add the external email, it becomes available for selection in the **Notification email** dropdown list.

Reviewer: Can we provide some examples of when this might be useful? For example something like

While external email notifications can be used to notify people outside of your dbt Cloud account, they can also be used to setup third party integrations that are not yet natively supported. For example:
- Send notifications to microsoft teams channels: https://support.microsoft.com/en-us/office/tip-send-email-to-a-channel-2c17dbae-acdf-4209-a761-b463bdaaa4ca#:~:text=Select%20More%20options,address%20into%20a%20new%20email.
- Send notifications to pagerduty: https://support.pagerduty.com/docs/email-integration-guide

    <Lightbox src="/img/docs/deploy/example-notification-external-email.png" width="50%" title="Example of the Notification email dropdown"/>

1. Select the **Environment** (in the dropdown) for the jobs you want to receive notifications about. 

1. Click **Edit** to configure the email notification settings.

    - If you're a **user**, choose one or more of the run statuses (**Succeeds**, **Fails**, **Is Canceled**) for each job you want to receive notifications for.

>     - If you're an **admin**, use the **Configure notifications for** dropdown to select all the users you want to sent notifications to. Then, choose one or more of the run statuses (**Succeeds**, **Fails**, **Is Canceled**) for each job you want them to receive notifications for.

Reviewer: I don't think this is totally accurate. You can only configure notifications for one user at a time, and the clicking edit disables the dropdown until the currently selected item has been saved

Reviewer: Can we also specify "account admin" here?

1. When you're done with the settings, click **Save**.
    
    To set up alerts on jobs from a different environment, select another **Environment** from the dropdown, **Edit** those job notification settings, and click **Save**.

    <Lightbox src="/img/docs/deploy/example-email-notification-settings-page.png" width="100%" title="Example of the Email notifications page"/>

### Unsubscribe from email notifications
1. From the gear menu in the upper right corner, choose **Notification settings**.
1. On the **Email notifications** page, click **Unsubscribe from all email notifications**. 

## Slack notifications

You can receive Slack alerts by setting up the Slack integration first, then configuring the dbt Cloud notification settings.

:::note 
If there have been changes to the user roles and you need to move ownership, please [contact support](mailto:support@getdbt.com) and provide the necessary information for them to make this change for you.

> :::

Reviewer: Maybe something liek this:

"While any account admin can edit slack notifications, they will be limited to configuring the channels to which their Slack user has access to. If there has been a change in user roles or slack permissions such that you no longer have access to edit a configured slack channel, please contact support"

Author: perfect!

1. Select **Slack notifications** in the left sidebar. 
1. Select the **Notification channel** (in the dropdown) you want to send the job run notifications to. 
    <Lightbox src="/img/docs/deploy/example-notification-slack-channels.png" width="75%" title="Example of the Notification channel dropdown"/>
1. Select the **Environment** (in the dropdown) for the jobs you want to receive notifications about. 
1. Click **Edit** to configure the Slack notification settings. Choose one or more of the run statuses (**Succeeds**, **Fails**, **Is Canceled**) for each job you want to receive notifications for.
1. When you're done with the settings, click **Save**.
    
    To send alerts to another Slack channel, select another **Notification channel** from the dropdown, **Edit** those job notification settings, and click **Save**.

    <Lightbox src="/img/docs/deploy/example-slack-notification-settings-page.png" width="100%" title="Example of the Slack notifications page"/>

### Disable the Slack integration

1. Go to your [Personal profile](https://cloud.getdbt.com/settings/profile) page and scroll to the **Linked accounts** section.

> 1. Find the Slack application in the **Linked accounts** section, click the trash can icon, and click **Unlink**. The Slack notification settings are no longer active.

Reviewer: So in this scenario, the slack notification will still show up for other admins, but will not work until one of those admins goes in and edits and re-saves the notification. This is because we send slack notifications using the slack token of the user who last updated the notification. I wonder if we can specify this somehow in less technical terms?

Author: added this bit. not sure if i understood you 100% tho. please take a look and lemme know!

---
title: "Job notifications"
id: "job-notifications"
description: "Set up notifications in dbt Cloud to receive email or Slack alerts about job run status."
---

Set up notifications in dbt Cloud to receive email or Slack alerts when a job run succeeds, fails, or is cancelled.

## Email notifications

> You can receive email alerts by configuring the dbt Cloud notification settings. As a **user**, you can set up email notifications for yourself. As an **admin**, you can set up notifications for yourself and other team members. Refer to [Users and licenses](/docs/cloud/manage-access/seats-and-users) for details on license types that are eligible for email notifications.

Author: are **user** and **admin** the actual role names in dbt Cloud that should be used on this page? 

i continued to use those words as it was already there on the page but it didn't seem to match up when i looked at the pag https://docs.getdbt.com/docs/cloud/manage-access/seats-and-users

Reviewer: I think if we wanted to be more explicit we could say something like tihs:
- As a *developer user*, you can setup email notifications for yourself
- As an *account admin*, you can setup email notifications for yourself and other team members on your account
- As an *it or read only user*, you can _recieve_ email notifications, but not set them up

Author: thank you!

### Email
You can receive email alerts by configuring the dbt Cloud notification settings. As a **user**, you can set up email notifications for yourself. As an **admin**, you can set up notifications for yourself and other team members. Refer to [Users and licenses](/docs/cloud/manage-access/seats-and-users) for details on license types that are eligible for email notifications.

> These are the following options for setting up email notifications. Refer to [Users and licenses](/docs/cloud/manage-access/seats-and-users) for info on license types eligible for email notifications. 

Author: i added a prerequisites section for the `Slack notifications` section below. 

are there prereqs for email notifications that also need to be added here?

1. When you're done with the settings, click **Save**.
    
    To set up alerts on jobs from a different environment, select another **Environment** from the dropdown, **Edit** those job notification settings, and click **Save**.

    <Lightbox src="/img/docs/deploy/example-email-notification-settings-page.png" width="100%" title="Example of the Email notifications page"/>

### Unsubscribe from email notifications
1. From the gear menu in the upper right corner, choose **Notification settings**.
1. On the **Email notifications** page, click **Unsubscribe from all email notifications**. 

## Slack notifications

<!-- <Snippet path="slack-notifications-config-steps" /> -->

You can receive Slack alerts by setting up the Slack integration first, then configuring the dbt Cloud notification settings.

:::note 
If there have been changes to the user roles and you need to move ownership, please [contact support](mailto:support@getdbt.com) and provide the necessary information for them to make this change for you.
:::

### Prerequisites 

> - You must be an administrator of the Slack workspace to configure Slack notifications in dbt Cloud. 

Author: are there other prereqs we should add to this list?

---
title: "Job notifications"
id: "job-notifications"
description: "Set up notifications in dbt Cloud to receive email or Slack alerts about job run status."
---

Set up notifications in dbt Cloud to receive email or Slack alerts when a job run succeeds, fails, or is cancelled.

## Email notifications

You can receive email alerts about jobs by configuring the dbt Cloud email notification settings.

### Prerequisites 
- You must be either a _developer user_ or an _account admin_ to configure email notifications in dbt Cloud. For more details, refer to [Users and licenses](/docs/cloud/manage-access/seats-and-users).
    - As a developer user, you can set up email notifications for yourself. 
    - As an account admin, you can set up notifications for yourself and other team members. 

### Configure email notifications

1. From the gear menu in the upper right corner, choose **Notification settings**.

> 1. By default, dbt Cloud sends notifications to the email address that's in your [User profile](https://cloud.getdbt.com/settings/profile) page.

Reviewer 2: We should avoid product links like this in the docs. Multi-cell architecture is coming soon and will break these (also to avoid our docs alienating emea and AU clients since this feature is available for them as well).
```suggestion
1. By default, dbt Cloud sends notifications to the email address in your User profile page.
```

Author: good callout!

>     If you're an account admin, you can choose a different email address to send notifications to. Select the **Notification email** dropdown and choose another address from the list. The list includes **Internal Users** with access to the account and also **External Emails** that have been added. 

Reviewer 2: ```suggestion
    If you're an account admin, you can choose a different email address to receive notifications. Select the **Notification email** dropdown and choose another address from the list. The list includes **Internal Users** with access to the account and **External Emails** that have been added. 
```

    - To add an external email address, select the **Notification email** dropdown and choose **Add external email**. After you add the external email, it becomes available for selection in the **Notification email** dropdown list. External emails can be addresses that are outside of your dbt Cloud account and also for third-party integrations like [channels in Microsoft Teams](https://support.microsoft.com/en-us/office/tip-send-email-to-a-channel-2c17dbae-acdf-4209-a761-b463bdaaa4ca) and [PagerDuty email integration](https://support.pagerduty.com/docs/email-integration-guide).

    <Lightbox src="/img/docs/deploy/example-notification-external-email.png" width="50%" title="Example of the Notification email dropdown"/>

> 1. Select the **Environment** (in the dropdown) for the jobs you want to receive notifications about. 

Reviewer 2: ```suggestion
1. Select the **Environment** for the jobs you want to receive notifications about from the dropdown.
```

1. Click **Edit** to configure the email notification settings. Choose one or more of the run statuses (**Succeeds**, **Fails**, **Is Canceled**) for each job you want to receive notifications about.

1. When you're done with the settings, click **Save**.

>     If there is another person you also want to send the notifications to, as an account admin, you can select another **Notification email** from the dropdown, **Edit** the job notification settings, and **Save** the changes.

Reviewer 2: ```suggestion
    As an account admin, you can add more recipients to the notifications by choosing another **Notification email** from the dropdown, **Edit** the job notification settings, and **Save** the changes.
```

    
    To set up alerts on jobs from a different environment, select another **Environment** from the dropdown, **Edit** those job notification settings, and **Save** the changes. 

    <Lightbox src="/img/docs/deploy/example-email-notification-settings-page.png" width="100%" title="Example of the Email notifications page"/>

### Unsubscribe from email notifications
1. From the gear menu in the upper right corner, choose **Notification settings**.
1. On the **Email notifications** page, click **Unsubscribe from all email notifications**. 

## Slack notifications

> You can receive Slack alerts about jobs by setting up the Slack integration, then configuring the dbt Cloud Slack notification settings.

Reviewer 2: Potentially save some space below. This may be better served as an individual page under setup as well
```suggestion
You can receive Slack notifications about jobs by setting up the [Slack integration](/faqs/Accounts/slack), then configuring the dbt Cloud Slack notification settings.
```

:::note 

> Any account admin can edit the Slack notifications but they'll be limited to configuring the channels that the Slack user has access to. If there has been a change in user roles or Slack permissions where you no longer have access to edit a configured Slack channel, please [contact support](mailto:support@getdbt.com) for assistance. 

Reviewer: Can we say "that their Slack user has access to"?

> Any account admin can edit the Slack notifications but they'll be limited to configuring the channels that the Slack user has access to. If there has been a change in user roles or Slack permissions where you no longer have access to edit a configured Slack channel, please [contact support](mailto:support@getdbt.com) for assistance. 

Author: ```suggestion
Any account admin can edit the Slack notifications but they'll be limited to configuring the channels that their Slack user has access to. If there has been a change in user roles or Slack permissions where you no longer have access to edit a configured Slack channel, please [contact support](mailto:support@getdbt.com) for assistance. 
```

:::

### Prerequisites 
- You must be an administrator of the Slack workspace. 
- You must be an account admin to configure Slack notifications in dbt Cloud. For more details, refer to [Users and licenses](/docs/cloud/manage-access/seats-and-users).

> ### Set up the Slack integration

Reviewer 2: Could potentially save space here and better serve customers by linking to the existing FAQ. Maybe even a standalone page in the setup. Linking in the pre-reqs could save some space as well.

Author: for a better reading experience, i actually consolidated this information so it's in one place instead of various places and avoid tab fatigue. in this pr, i removed the FAQ and added a redirect to this page.

1. Go to your [User profile](https://cloud.getdbt.com/settings/profile) page and scroll to the **Linked accounts** section.
1. In the **Linked accounts** section, find the Slack application and click **Link**.
   <Lightbox src="/img/docs/dbt-cloud/Link-your-Slack-Profile.png" width="75%" title="Link for the Slack app"/>
1. Allow dbt Labs to access the Slack workspace. If you are a member of multiple workspaces, you can select the appropriate workspace from the dropdown menu in the upper right corner.
   <Lightbox src="/img/docs/dbt-cloud/Allow-dbt-to-access-slack.png" width="75%" title="Allow dbt access to Slack"/>

### Configure Slack notifications

1. From the gear menu in the upper right corner, choose **Notification settings**. 
1. Select **Slack notifications** in the left sidebar. 

> 1. Select the **Notification channel** (in the dropdown) you want to send the job run notifications to. 

Reviewer 2: ```suggestion
1. Select the **Notification channel** you want to receive the job run notifications from the dropdown. 
```

    <Lightbox src="/img/docs/deploy/example-notification-slack-channels.png" width="75%" title="Example of the Notification channel dropdown"/>
1. Select the **Environment** (in the dropdown) for the jobs you want to receive notifications about. 
1. Click **Edit** to configure the Slack notification settings. Choose one or more of the run statuses (**Succeeds**, **Fails**, **Is Canceled**) for each job you want to receive notifications about.
1. When you're done with the settings, click **Save**.
    
    To send alerts to another Slack channel, select another **Notification channel** from the dropdown, **Edit** those job notification settings, and **Save** the changes.

    To set up alerts on jobs from a different environment, select another **Environment** from the dropdown, **Edit** those job notification settings, and **Save** the changes.

    <Lightbox src="/img/docs/deploy/example-slack-notification-settings-page.png" width="100%" title="Example of the Slack notifications page"/>

### Disable the Slack integration

1. Go to your [User profile](https://cloud.getdbt.com/settings/profile) page and scroll to the **Linked accounts** section.

> 1. Find the Slack application in the **Linked accounts** section, click the trash can icon, and click **Unlink**. You will no longer receive Slack notifications. However, other account admins might still receive the notifications. If they want to stop receiving the notifications, they can update their [Slack notification configuration](#configure-slack-notifications) or disable the Slack integration on their dbt Cloud account. 

Reviewer: For the second bullet, how about something like the following? I think it's a bit more accurate:

> Find the Slack application in the **Linked accounts** section, click the trash can icon, and click **Unlink**. Channels **you configured** will no longer receive Slack notifications. Channels configured by other users will continue to receive Slack notifications as long as those users still have active Slack integrations. To migrate ownership of a Slack channel notification configuration, have another account admin edit the configuration.

Author: thank you!

> 1. Find the Slack application in the **Linked accounts** section, click the trash can icon, and click **Unlink**. You will no longer receive Slack notifications. However, other account admins might still receive the notifications. If they want to stop receiving the notifications, they can update their [Slack notification configuration](#configure-slack-notifications) or disable the Slack integration on their dbt Cloud account. 

Author: ```suggestion
1. Find the Slack application in the **Linked accounts** section, click the trash can icon, and click **Unlink**. Channels that you configured will no longer receive Slack notifications. Channels that are configured by other account admins will continue to receive Slack notifications if they still have active Slack integrations. To migrate ownership of a Slack channel notification configuration, have another account admin edit the configuration.
```
